packages/components/ng-ui-router-layout/src/oui-modal/index.js

```diff
@@ -153,8 +153,17 @@ angular
         };
 
         // close state modal when exiting the state
-        state.onExit = () => {
+        state.onExit = (transition) => {
           modalInstance.close();
+          // call state declaration onExit function if present
+          let onExitFn = state.self.onExit;
+          if (isArray(onExitFn)) {
```

Comment:
No, in fact, you have some injections. Once your state declared, the onExit function becomes an array if you have params in declaration. The last index is the function itself and the others indexes are the params that will be injected.
It's like you define an angular service:
```
angular.service('myService', ($log, $q) => { ... });
```

This will be annoted to something like:
```
angular.service('myService', ['$log', '$q', function ($log, $q) { ... }];
```

That's what is done by babel-plugin-angularjs-annotate and how angularJS worked at the beginning :-)